net: config: align Kconfig debug levels with OpenThread ones

edmont opened the pull request:
OpenThread moved from 4 to 5 debug levels and it was not possible to configure all of them with Kconfig.

Reference: https://github.com/zephyrproject-rtos/openthread/blob/zephyr/include/openthread/platform/logging.h#L45-L107

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

rlubos commented:
Looks good, but shall we also add the NOTE log level handling in `otPlatLog` function?
https://github.com/zephyrproject-rtos/zephyr/blob/2716cbcaa6599374d0e80c473207e43271ad6968/subsys/net/lib/openthread/platform/logging.c#L40

edmont committed:
net: config: align Kconfig debug levels with OpenThread ones

OpenThread moved from 4 to 5 debug levels and it was not possible
to configure all of them with Kconfig.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

edmont commented:
That's right. Thanks @rlubos

rlubos reviewed (approved):
Thanks, LGTM.
